src/structs/jpeg_header.rs

```diff
@@ -871,6 +904,27 @@ impl JPegHeader {
         }
         return Ok(ParseSegmentResult::Continue);
     }
+
+    /// constructs the quantization table based on the jpeg header
+    pub fn construct_quantization_table(&self) -> Result<Vec<QuantizationTables>> {
```

Comment:
Slightly misleading name
```suggestion
    pub fn construct_quantization_tables(&self) -> Result<Vec<QuantizationTables>> {
```